packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -168,14 +197,100 @@ contract L1Block is ISemver, IGasToken {
+        } else if (_type == Types.ConfigType.SET_L1_CROSS_DOMAIN_MESSENGER_ADDRESS) {
+            data_ = abi.encode(Storage.getAddress(L1_CROSS_DOMAIN_MESSENGER_ADDRESS_SLOT));
+        } else if (_type == Types.ConfigType.SET_L1_STANDARD_BRIDGE_ADDRESS) {
+            data_ = abi.encode(Storage.getAddress(L1_STANDARD_BRIDGE_ADDRESS_SLOT));
+        } else if (_type == Types.ConfigType.SET_SEQUENCER_FEE_VAULT_CONFIG) {
+            data_ = abi.encode(Storage.getBytes32(SEQUENCER_FEE_VAULT_CONFIG_SLOT));
+        } else if (_type == Types.ConfigType.SET_L1_FEE_VAULT_CONFIG) {
+            data_ = abi.encode(Storage.getBytes32(L1_FEE_VAULT_CONFIG_SLOT));
+        }
+    }
+
     /// @notice Sets the gas paying token for the L2 system. Can only be called by the special
-    ///         depositor account. This function is not called on every L2 block but instead
-    ///         only called by specially crafted L1 deposit transactions.
+    ///         depositor account, initiated by a deposit transaction from L1.
+    ///         This is a legacy setter that exists to give compabilitity with the legacy SystemConfig.
+    ///         Custom gas token can now be set using `setConfig`. This can be removed in the future.
     function setGasPayingToken(address _token, uint8 _decimals, bytes32 _name, bytes32 _symbol) external {
         if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
 
         GasPayingToken.set({ _token: _token, _decimals: _decimals, _name: _name, _symbol: _symbol });
-
         emit GasPayingTokenSet({ token: _token, decimals: _decimals, name: _name, symbol: _symbol });
     }
+
+    /// @notice
+    function setHolocene() external {
+        if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
+        // TODO set holocene activation to true
+
+        Storage.setBytes32(BASE_FEE_VAULT_CONFIG_SLOT, _migrateFeeVaultConfig(Predeploys.BASE_FEE_VAULT));
+        Storage.setBytes32(L1_FEE_VAULT_CONFIG_SLOT, _migrateFeeVaultConfig(Predeploys.L1_FEE_VAULT));
+        Storage.setBytes32(SEQUENCER_FEE_VAULT_CONFIG_SLOT, _migrateFeeVaultConfig(Predeploys.SEQUENCER_FEE_WALLET));
+        Storage.setAddress(
+            L1_CROSS_DOMAIN_MESSENGER_ADDRESS_SLOT,
+            address(IL1CrossDomainMessenger(Predeploys.L2_CROSS_DOMAIN_MESSENGER).otherMessenger())
+        );
+        Storage.setAddress(
+            L1_STANDARD_BRIDGE_ADDRESS_SLOT,
+            address(IStandardBridge(payable(Predeploys.L2_STANDARD_BRIDGE)).otherBridge())
+        );
+        Storage.setAddress(
+            L1_ERC_721_BRIDGE_ADDRESS_SLOT, address(IERC721Bridge(Predeploys.L2_ERC721_BRIDGE).otherBridge())
+        );
+        Storage.setUint(
+            REMOTE_CHAIN_ID_SLOT, IOptimismMintableERC721Factory(Predeploys.L2_ERC721_BRIDGE).remoteChainId()
+        );
+    }
+
+    /// @notice Helper function for migrating deploy config.
+    ///         TODO: may need to make 3 calls for legacy purposes, the config abi doesn't yet exist
```

Comment:
This is important, we cannot use the `config()` function as it will not exist when this migration is happening